ha/core/system_health/system_health.py

```diff
@@ -402,41 +427,72 @@ def process_event(self, healthevent: HealthEvent):
+                latest_health = EntityHealth.read(current_health)
+                if stored_genration_id != incoming_generation_id:
+                    if incoming_health_status == status:
+                        # If the generation id matches and stored node health matches
+                        # with incoming node health, means online event received first
+                        # instead of failed event in delete scenario
+                        healthevent.specific_info = {"generation_id": stored_genration_id, "pod_restart": 1}
+                        updated_health = SystemHealth.create_updated_event_object(healthevent.timestamp, current_timestamp, "failed", healthevent.specific_info, latest_health)
+                        # Create a "failed" event and update it in system health and publish
+                        self._check_and_update(current_health, updated_health, healthevent, next_component)
+                        current_health = updated_health
+                        # Now create an "online" event and update it in system health and publish
+                        healthevent.specific_info = {"generation_id": incoming_generation_id, "pod_restart": 1}
+                        updated_health = SystemHealth.create_updated_event_object(healthevent.timestamp, current_timestamp, status, healthevent.specific_info, latest_health)
+                        self._check_and_update(current_health, updated_health, healthevent, next_component)
+                    elif pod_restart_val is not None and pod_restart_val:
+                        # Check the pod_restart value assosciated with Node, if its 1,
+                        # means this alert is already updated. No need to send the alert again.
+                        # Just need to reset the pod_restart value
+                        key = self._prepare_key(component, cluster_id=self.node_map['cluster_id'], \
+                                site_id=self.node_map['site_id'], rack_id=self.node_map['rack_id'], \
+                                node_id=self.node_id)
+                        latest_health_dict = json.loads(current_health)
+                        new_spec_info = {"generation_id": stored_genration_id, "pod_restart": 0}
+                        latest_health_dict["events"][0]["specific_info"] = new_spec_info
+                        updated_health = EntityHealth.write(latest_health_dict)
+                        self.healthmanager.set_key(key, updated_health)
+                else:
+                    # current health is there and generation id is also already present.
+                    # That means its a normal failure scenario
+                    updated_health = SystemHealth.create_updated_event_object(healthevent.timestamp, current_timestamp, status, healthevent.specific_info, latest_health)
+                    self._check_and_update(current_health, updated_health, healthevent, next_component)
+            elif current_health and not specific_info:
```

Comment:
When would this scenario occur ?

Reply:
Its not a scenario. It is to update hierarchical components such as site and rack where specific_info is not get stored. But, the place of this elif is not correctly placed. Updating in the latest patch.